### PR TITLE
feat: parseJSON cheatcodes with type coercion

### PR DESCRIPTION
**Pull request by odyslam**
Should be merged after https://github.com/foundry-rs/foundry/pull/4061

### CODE REVIEW

**Comment by odyslam** [comment]
Failed test seems irrelevant to the PR

File: src/Vm.sol [hunk 1]
```diff
@@ -33,137 +33,210 @@ interface VmSafe {
 
     // Loads a storage slot from an address
     function load(address target, bytes32 slot) external view returns (bytes32 data);
+
```
Comment (mds1) [inline src/Vm.sol:36]: Why were all these blank lines introduced? We should undo that

File: src/Vm.sol [hunk 2]
```diff
@@ -183,50 +256,97 @@ interface VmSafe {
...
     function parseJson(string calldata json) external pure returns (bytes memory abiEncodedData);
 
+    // The following parseJson cheatcodes will do type coercion, for the type that they indicate.
+    // For example, parseJsonUint will coerce all values to a uint256. That includes stringified numbers '12'
+    // and hex numbers '0xEF'.
+    // Type coercion works ONLY for discrete values or arrays. That means that the key must return a value or array, not
+    // a JSON object.
+
+    function parseJsonUint(string calldata, string calldata) external returns (uint256);
+
```
Comment (mds1) [inline src/Vm.sol:269]: And we should remove the blank lines between all these new cheats too

Comment (odyslam) [reply src/Vm.sol:269]: This is the result of running `forge fmt`. It's more legible this way, no?

Comment (mds1) [reply src/Vm.sol]: Maybe you're on an old forge version or have some custom config? But for me and in CI `forge fmt` doesn't introduce line breaks. Just pushed dac8350 to match the original formatting for now to reduce the diff (give that a review to make sure I didn't accidentally remove anything, then we can merge). Open to organizing/cleaning up the `Vm` file in a separate issue/PR

Comment (odyslam) [reply src/Vm.sol]: @mds1 uuuh, didn't think about that. Probs you are right. lgtm

**Comment by odyslam** [comment]
LGTM

**Review by mds1 — APPROVED** [review]
Thanks!

**Pull request merged**